front/src/applications/stdcmV2/components/StdcmVias.tsx

```diff
@@ -48,8 +48,8 @@ const StdcmVias = ({ disabled = false, setCurrentSimulationInputs }: StdcmConfig
     dispatch(deleteViaV2(index));
   };
 
-  const onClickOnCard = () => {
-    const newPathSteps = addElementAtIndex(pathSteps, pathSteps.length - 1, null);
+  const onClickOnCard = (pathStepId: number) => {
```

Comment:
Maybe addPathStep or addVia would be a better name ? Or addViaOnClick ?
```suggestion
  const addPathStep = (pathStepId: number) => {
```